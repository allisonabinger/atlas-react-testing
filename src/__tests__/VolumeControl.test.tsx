import { render, fireEvent } from "@testing-library/react";
import { expect, test, describe } from "vitest";
import VolumeControl from "../components/VolumeControl";

describe("Volume Control Component", () => {
  test("renders correctly and matches snapshot", () => {
    const { container } = render(<VolumeControl />);
    expect(container).toMatchSnapshot();
  });

  test("matches snapshot when muted", () => {
    const { container, getAllByRole } = render(<VolumeControl />);
    const buttons = getAllByRole("button");
    const muteButton = buttons[0];

    fireEvent.click(muteButton);
    expect(container).toMatchSnapshot();
  });

  test("matches snapshot at a different volume", () => {
    const { container, getAllByRole } = render(<VolumeControl />);

    const sliders = getAllByRole("slider");
    const volumeSlider = sliders[0];

    fireEvent.change(volumeSlider, {
      target: {
        value: "50",
      },
    });
    expect(container).toMatchSnapshot();
  });
});
