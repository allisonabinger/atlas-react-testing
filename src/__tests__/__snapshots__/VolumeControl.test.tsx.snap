// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Volume Control Component > matches snapshot at a different volume 1`] = `
<div>
  <div
    class="flex items-center justify-center"
  >
    <button
      class="volume"
    >
      <img
        alt="Volume Icon"
        class="size-7"
        src="/src/assets/volume.svg"
      />
    </button>
    <input
      class="w-full h-8 accent-vista-blue-700 outline-2 outline-transparent outline-offset-2"
      id="volumeSlider"
      max="100"
      min="0"
      type="range"
      value="75"
    />
  </div>
</div>
`;

exports[`Volume Control Component > matches snapshot when muted 1`] = `
<div>
  <div
    class="flex items-center justify-center"
  >
    <button
      class="volume"
    >
      <img
        alt="Volume Icon"
        class="size-7"
        src="/src/assets/volume.svg"
      />
    </button>
    <input
      class="w-full h-8 accent-vista-blue-700 outline-2 outline-transparent outline-offset-2"
      id="volumeSlider"
      max="100"
      min="0"
      type="range"
      value="75"
    />
  </div>
</div>
`;

exports[`Volume Control Component > renders correctly and matches snapshot 1`] = `
<div>
  <div
    class="flex items-center justify-center"
  >
    <button
      class="volume"
    >
      <img
        alt="Volume Icon"
        class="size-7"
        src="/src/assets/volume.svg"
      />
    </button>
    <input
      class="w-full h-8 accent-vista-blue-700 outline-2 outline-transparent outline-offset-2"
      id="volumeSlider"
      max="100"
      min="0"
      type="range"
      value="75"
    />
  </div>
</div>
`;
